### PR TITLE
No dict update

### LINKED ISSUE #1228: update_forward_refs modifies __dict__ of a module
# Bug

```
OS: macOS Catalina 10.15.3  
python: 3.7.3
pydantic version: 1.1
```

After creating a BaseModel subclass and calling `update_forward_ref()` classmethod, pydantic can modify __dict__ of a module in which the model was created if the model was declared inside some function of this module. 
```py

from typing import Optional, ForwardRef
from pydantic import BaseModel

def first_func():
    class Mod(BaseModel):
        field: Optional[ForwardRef("Mod")]
    Mod.update_forward_refs()
    return Mod

def second_func():
    class Mod(BaseModel):
        another_field: Optional[ForwardRef("Mod")]

    Mod.update_forward_refs()
    
    return Mod

first_model = first_func()
second_model = second_func()

print(second_model.__fields__["another_field"].type_)
# Output: <class '__main__.first_func.<locals>.Mod'>
# Expected: <class '__main__.second_func.<locals>.Mod'>
```

It happens because of these two lines: https://github.com/samuelcolvin/pydantic/blob/645e5fe6a0c74c95c24410571e9bb804af3eb677/pydantic/main.py#L632-L633  
They can be fixed that way:  
```py
        globalns = sys.modules[cls.__module__].__dict__.copy()
        globalns.setdefault(cls.__name__, cls)
```

This bug can lead to errors when two different models with the same names are defined in two different test cases or some other errors which can be hard to debug.

### PR DESCRIPTION
**Pull request by Paul-Ilyin**
## Change Summary
fixes #1228 

## Checklist

* [x] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

**Comment by samuelcolvin** [comment]
Could you look at why tests are failing?

Also, is it possible to add a test to lock in this change?

**Comment by Paul-Ilyin** [comment]
I've added a unit test. I'm not sure, but I guess that the coverage change is because codecov compares the PR with 1 year old commit

File: tests/test_aliases.py [hunk 1]
```diff
@@ -330,10 +330,7 @@ class Child(Parent):
         a: str = Field(..., alias='a_field_child')
 
         class Config:
-            fields = {
-                'a': dict(alias='a_config_child'),
-                'b': dict(alias='b_config_child'),
-            }
+            fields = {'a': dict(alias='a_config_child'), 'b': dict(alias='b_config_child')}
```
Comment (samuelcolvin) [inline tests/test_aliases.py]: this is black being weird, please can you revert these and `test_decorator.py`

**Comment by samuelcolvin** [comment]
thanks so much.

**Pull request merged**